### PR TITLE
Add `Error.metadata/2` to extract error metadata

### LINKED ISSUE #357: Support Erlang error messages better
Now we get an error message as the error class, which will be truncated if it's longer than 100 characters.

```
{:badkey, :message, %Ecto.InvalidChangesetError{action: :update, changeset: #Ecto.Changeset<action:
```

Would be nice to unwrap an Erlang error message into something smaller, such as `:badkey`, or if available -> the Elixir exception struct.

As reported in: https://app.intercom.io/a/apps/yzor8gyw/inbox/inbox/conversation/15857926961

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
Closes #357.

Instead of extracting error metadata manually in `ErrorHandler.extract_reason_and_message/2`, this patch adds `Error.metadata/2`, which will be used to extract metadata for all errors. 

It uses `Exception.normalize` causing erlang errors to be converted to `ErlangError`s. This prevents unsanitized error tuples from being added to transactions.

### CODE REVIEW

**Comment by jeffkreeftmeijer** [comment]
[appsignal-phoenix-example](https://github.com/jeffkreeftmeijer/appsignal-phoenix-example) now has an ExceptionController with some exceptions to test with on http://localhost:4000/exceptions:

- raise
- exit
- 404
- proc_raise
- proc_exit
- timeout
- template
- template_error

The 404, which links to a path not present in the router, raises a `Phoenix.Router.NoRouteError`, which should be reported as such through the ErrorHandler.

The changes in https://github.com/appsignal/appsignal-elixir/pull/425 will make sure that error isn’t reported at all, as it has a < 500 status code, resolving https://github.com/appsignal/appsignal-elixir/issues/417.

File: lib/appsignal/plug.ex [hunk 1]
```diff
@@ -48,26 +48,25 @@ if Appsignal.plug?() do
       handle_error(conn, kind, reason, reason, stack)
     end
 
+    def handle_error(_conn, kind, %{plug_status: status} = reason, _wrapped_reason, stack)
+        when status < 500 do
```
Comment (tombruijn) [inline lib/appsignal/plug.ex]: what's the behavior here? We do catch errors below code 500 now, right? so why do we need to reraise it?

Comment (jeffkreeftmeijer) [reply lib/appsignal/plug.ex]: We don't in this patch. After all three are merged, the behaviour should be the same as before, without < 500 errors being caught by the ErrorHandler and ending up in AppSignal (#417). We'll make a decision on more flexible filtering in #410.

**Comment by jeffkreeftmeijer** [comment]
Added `TransactionRegistry.ignore/1` and `.ignored?/1` to explicitly ignore further errors on processes. It's used in `Appsignal.Plug` to ignore errors we've sent an error for already. Since this is explicit and used in the Plug integration, this implementation won't run into #299 again.

**Comment by tombruijn** [comment]
Can it happen that another error occurs and because it's ignored we miss the error from the process thereafter? E.g. a `send_error` call?

File: lib/appsignal/plug.ex (hunk 1 above)
Comment (tombruijn) [reply lib/appsignal/plug.ex]: Ok, so we don't want to fix that issue (not tracking non-500 status code) errors in these PRs just yet? Just want to make sure

Comment (jeffkreeftmeijer) [reply lib/appsignal/plug.ex]: That's correct. We'll tackle that in #410.

**Pull request merged**